core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1179,7 +1179,9 @@ private SqlCall toSql(SqlOperator op, boolean distinct,
 
       // Handle filter by generating FILTER (WHERE ...)
       final SqlCall call2;
-      if (filterArg < 0) {
```

Comment:
Why cannot we just check if `op instanceof SqlOperatorTable.APPROX_COUNT_DISTINCT`